Update test to be compatible with latest Glamour v0.8.0

ottok opened the pull request:
Latest Glamour has slightly changed logic in line length / wrapping, resulting test failures due to string mismatch. Update tests and bump dependency to v0.8.0.

This was detected then building the GitHub cli pacakge `gh` in Debian started to fail with src:golang-github-charmbracelet-glamour 0.8.0-1.

ottok committed:
Update test to be compatible with latest Glamour v0.8.0

Latest Glamour has slightly changed logic in line length / wrapping,
resulting test failures due to string mismatch. Update tests and bump
dependency to v0.8.0, and others to the bare minimal level as generated
by `go mod tidy`.

This was detected then building the GitHub cli package `gh` in Debian
started to fail with src:golang-github-charmbracelet-glamour 0.8.0-1.

Closes: #10179

cliAutomation commented:
Hi! Thanks for the pull request. Please ensure that this change is linked to an issue by mentioning an issue number in the description of the pull request. If this pull request would close the issue, please put the word 'Fixes' before the issue number somewhere in the pull request body. If this is a tiny change like fixing a typo, feel free to ignore this message.

jtmcg commented:
Hey @ottok, thanks for the submission. It's not entirely clear to me where these failures are happening, as none of the tests are failing locally for me. Could you [open an issue](https://github.com/cli/cli/issues/new/choose) to link to this and describe the repro steps? Ultimately, I'd like to understand what's going on so we can catch it before release in the future.

ottok commented:
Hi!

The original error was reported in Debian in https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1091585 and fixed in https://salsa.debian.org/go-team/packages/gh/-/commit/8faccf149d0836f8ff285808182c15cf02dd9f28 that was included in the latest upload the Debian unstable.

Here is an example of one of the failures for reference:
```
=== RUN   TestPRReview_interactive
>     review_test.go:283: 
>         	Error Trace:	/<<PKGBUILDDIR>>/_build/src/github.com/cli/cli/v2/pkg/cmd/pr/review/review_test.go:283
>         	Error:      	Not equal: 
>         	            	expected: "Got:\n\n  cool story                                                                  \n\n"
>         	            	actual  : "Got:\n\n  cool story                                                                      \n\n"
>         	            	
>         	            	Diff:
>         	            	--- Expected
>         	            	+++ Actual
>         	            	@@ -2,3 +2,3 @@
>         	            	 
>         	            	-  cool story                                                                  
>         	            	+  cool story                                                                      
>         	            	 
>         	Test:       	TestPRReview_interactive
> --- FAIL: TestPRReview_interactive (0.00s)
```

I already put in a sizeable amount of effort into fixing this and submit upstream. It feels that your request to file a bug report about this is mostly process work and not very productive, so I will pass on it. Take this or close it and re-implement it yourself in the future when you decide to upgrade Glamour.

jtmcg commented:
> I already put in a sizeable amount of effort into fixing this and submit upstream. It feels that your request to file a bug report about this is mostly process work and not very productive, so I will pass on it.

Hey @ottok, while I understand this feeling, there's a few reasons we would appreciate a bug report beyond "process work"

1. Issues are more discoverable than PRs for the community. This helps prevent other folks from putting a sizeable amount of effort into opening duplicate issues or implementing PRs that others have already put a sizeable amount of effort into. If wasting your time is your primary concern for not opening an issue, consider that that choice may result in others wasting time instead.
2. Issues are our preferred form for facilitating conversations and understanding about what problems our users may be facing and aligning on a proposed fix. I, for one, am very unfamiliar with Debian, how these bugs are reported/addressed, and why we may need to upgrade Glamour. I would find value in the discussion that would follow in the issue. If I have questions, it is likely others do too, and having that conversation on an issue helps avoid the need to ask those questions again in the future.
3. Our issue templates are designed to reduce the amount of back and forth necessary for maintainers and contributors to understand what is going on

Ultimately, my goal is to understand what is going on, but you have seemed to reject that premise as process. We have a lot of contributors to our tool, and try to do our best by them, but sometimes that requires a little help from them, too. I hope this demonstrates why my ask isn't just unnecessary process work.

ottok commented:
Ok, as you spent that much effort in asking me to do it, I filed now #10179.

I also rebased this on latest trunk, and amended commit message with "Closes: #10179" so when you merge this, the bug report will auto-close.

williammartin commented:
@ottok, it looks like I can't push to your branch to fix the merge conflicts. It should just be a matter of removing the go mod changes and go getting the new version of glamour, rather than fighting the conflicts themselves. Happy to merge when that is done, sorry for the busy work.

ottok commented:
I can rebase this today if you are now ready to merge.

williammartin commented:
I'm ready to merge.

ottok commented:
I dropped the existing go.{mod,sum} modifications, added manually the update `github.com/charmbracelet/glamour v0.7.0 -> v0.8.0` and then ran `go mod tidy` go get the indirect dependencies updated automatically.

williammartin commented:
Thanks for doing the work on this.

andyfeller commented:
@jtmcg @williammartin : are there any concerns about [`cli/go-gh` still using the older versions of `lipgloss` and `glamour`](https://github.com/cli/go-gh/blob/13104ed7b2db4b8c1a83de4248ddfaaab7682916/go.mod#L8-L9)?

```
module github.com/cli/go-gh/v2

go 1.21

require (
	github.com/AlecAivazis/survey/v2 v2.3.7
	github.com/MakeNowJust/heredoc v1.0.0
	github.com/charmbracelet/glamour v0.7.0
	github.com/charmbracelet/lipgloss v0.10.1-0.20240413172830-d0be07ea6b9c
```

williammartin commented:
None that I can think of. Would be fine to update them as well though.
